src/libraries/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -171,6 +172,57 @@ private static bool TryDecodeUtf8(ReadOnlySpan<byte> input, out string decoded)
             return false;
         }
 
+        public string GetHeaderValue(string headerValue)
+        {
+            if (headerValue.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            // If it's a known header value, use the known value instead of allocating a new string.
+            if (_knownHeader != null)
+            {
+                if (_knownHeader.KnownValues != null)
+                {
+                    string[] knownValues = _knownHeader.KnownValues;
+                    for (int i = 0; i < knownValues.Length; i++)
+                    {
+                        if (knownValues[i].Equals(headerValue, StringComparison.OrdinalIgnoreCase))
+                        {
+                            return knownValues[i];
+                        }
+                    }
+                }
+
+                if (_knownHeader == KnownHeaders.Location)
```

Comment:
this path is not accessible from HTTP2 and for HTTP1.1 there is an original method. I was about to remove this branch